jsonrpc/metrics.go

```diff
@@ -0,0 +1,65 @@
+package jsonrpc
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	prometheus "github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics represents the jsonrpc metrics
+type Metrics struct {
+	// Requests number
+	Requests metrics.Counter
+
+	// Errors number
+	Errors metrics.Counter
+
+	// Requests duration (seconds)
+	RequestDurationSeconds metrics.Histogram
+}
+
+// GetPrometheusMetrics return the blockchain metrics instance
+func GetPrometheusMetrics(namespace string, labelsWithValues ...string) *Metrics {
+	labels := []string{}
+
+	for i := 0; i < len(labelsWithValues); i += 2 {
+		labels = append(labels, labelsWithValues[i])
+	}
+
+	return &Metrics{
+		Requests: prometheus.NewCounterFrom(stdprometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "jsonrpc",
+			Name:      "requests",
+			Help:      "Requests number",
+		}, labels).With(labelsWithValues...),
+		Errors: prometheus.NewCounterFrom(stdprometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "jsonrpc",
+			Name:      "request_errors",
+			Help:      "Request errors number",
+		}, labels).With(labelsWithValues...),
+		RequestDurationSeconds: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: "jsonrpc",
+			Name:      "request_duration_seconds",
+			Help:      "Request duration (seconds)",
+			Buckets: []float64{
```

Comment:
A little more buckets would make it more reasonable?
`0.001, 0.01, 0.1, 0.5, 1.0, 2.0`
Bucket `0.99` v.s. `1.0` isn't good enough.